python/cuvs_bench/cuvs_bench/run/__main__.py

```diff
@@ -0,0 +1,216 @@
+    "Any run groups found in the specified file/directory will "
+    "automatically override groups of the same name present in the "
+    "default configurations, including `base`.",
+)
+@click.option(
+    "--dataset",
+    default="glove-100-inner",
+    show_default=True,
+    prompt="Enter the name of dataset",
+    help="Name of dataset",
+)
+@click.option(
+    "--dataset-path",
+    default=lambda: os.environ.get(
+        "RAPIDS_DATASET_ROOT_DIR",
+        os.path.join(Path(__file__).parent, "datasets/"),
+    ),
+    show_default=True,
+    prompt="Enter the path to dataset folder",
+    help="Path to dataset folder, by default will look in "
+    "RAPIDS_DATASET_ROOT_DIR if defined, otherwise a datasets "
+    "subdirectory from the calling directory.",
+)
+@click.option("--build", is_flag=True, help="Build the index")
+@click.option("--search", is_flag=True, help="Perform the search")
+@click.option(
+    "--algorithms",
+    default="cuvs_cagra",
+    show_default=True,
+    prompt="Enter the comma separated list of named algorithms to run",
+    help="Run only comma separated list of named algorithms. If parameters "
+    "`groups` and `algo-groups` are both undefined, then group `base` "
+    "is run by default.",
+)
+@click.option(
+    "--groups",
+    default="base",
+    show_default=True,
+    prompt="Enter the comma separated groups of parameters",
+    help="Run only comma separated groups of parameters",
+)
+@click.option(
+    "--algo-groups",
+    help="Add comma separated <algorithm>.<group> to run. Example usage: "
+    ' "--algo-groups=raft_cagra.large,hnswlib.large".',
+)
+@click.option(
+    "-f",
+    "--force",
+    is_flag=True,
+    help="Re-run algorithms even if their results already exist",
+)
+@click.option(
+    "-m",
+    "--search-mode",
+    default="latency",
+    show_default=True,
+    prompt='Enter the search mode ("latency" or "throughput")',
+    help="Run search in 'latency' (measure individual batches) or "
+    "'throughput' (pipeline batches and measure end-to-end) mode.",
+)
+@click.option(
+    "-t",
+    "--search-threads",
+    default=None,
+    show_default=True,
+    help="Specify the number threads to use for throughput benchmark. "
+    "Single value or a pair of min and max separated by ':'. "
+    "Example: --search-threads=1:4. Power of 2 values between 'min' "
+    "and 'max' will be used. If only 'min' is specified, then a single "
+    "test is run with 'min' threads. By default min=1, "
+    "max=<num hyper threads>.",
+)
+@click.option(
+    "-r",
+    "--dry-run",
+    is_flag=True,
+    help="Dry-run mode will convert the yaml config for the specified "
+    "algorithms and datasets to the json format that’s consumed "
+    "by the lower-level c++ binaries and then print the command to "
+    "run execute the benchmarks but will not actually execute "
+    "the command.",
+)
+@click.option(
+    "--raft-log-level",
```

Comment:
Do references to RAFT need to be changed here, or do these still pass through to RAFT? This applies in several places.

Reply:
Yes this still passes through to RAFT